apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/instructorAiGenerateDraftEditor.html.tsx

```diff
@@ -171,6 +171,28 @@ export function InstructorAiGenerateDraftEditor({
   `.toString();
 }
 
+/**
+ * Returns the index of the prompt that represents the latest revision that would have been shown
+ * to the user after the prompt at the specified index. This specifically addresses the case where
+ * one or more `auto_revision` prompts follow an `initial` or `human_revision` prompt.
+ */
+function findLatestRevisionIndexForPrompt(
```

Comment:
What happens if the currentIndex is an auto_revision and no suitable next index is found? Should that throw an error.